pkg/util/pki/csr.go

```diff
@@ -178,7 +180,11 @@ func BuildCertManagerKeyUsages(ku x509.KeyUsage, eku []x509.ExtKeyUsage) []v1.Ke
 // The CSR will not be signed, and should be passed to either EncodeCSR or
 // to the x509.CreateCertificateRequest function.
 func GenerateCSR(crt *v1.Certificate) (*x509.CertificateRequest, error) {
-	commonName := crt.Spec.CommonName
+	commonName, err := extractCommonName(crt.Spec)
```

Comment:
I'm not really happy with the parsing of the literal subject twice. However, the alternative is to return a (optionally nil) RDNSequence from `extractCommonName` and carry it down to the CSR generation. Or the branching on the  literal subject should be done earlier.

Reply:
I think that this is OK for now 👍🏼 
Agree it's not elegant, but I guess it shouldn't be too expensive? And, we can change it later, there is a lot of complexity in how the two feature gates interact already + the existing subject vs literal subject- I feel like there will likely be more iterations anyway when this graduates to beta